### PR TITLE
Router refactor: Render Assessment > Cycle > Country

### PR DESCRIPTION
**Pull request by sorja**
Part 2 of #2937

chrome_v60Sj23QRb

- [x] Define and Render Assessment route
- [x] Define and Render Cycle route
- [x] Define and Render Country route

### CODE REVIEW

File: src/client/components/PageLayout/PageLayout.tsx [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-  const { children, withHeader, withToolbar } = props
+  const { withHeader, withToolbar } = props
+  useRedirect()
 
   return (
     <>
       {withHeader && <Header />}
       {withToolbar && <Toolbar />}
-      {children}
+      {/* {children} */}
```
Comment (minotogna) [inline src/client/components/PageLayout/PageLayout.tsx]: this comment can be removed

File: src/client/pages/PageRoutes/hooks/useRoutes.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+              {/*
+              AssessmentHome
+              AssessmentPrint
+              AssessmentDataDownload
+              Geo
+              DataExport
+              AssessmentSection
+              OriginalDataPoint
+              User
+              */}
```
Comment (minotogna) [inline src/client/pages/PageRoutes/hooks/useRoutes.tsx]: can we remove this comments? later review will be easier

File: src/client/pages/PageRoutes/PageRoutes.tsx [hunk 3]
```diff
@@ -22,14 +20,15 @@ const PageRoutes: React.FC = () => {
...
-        <Route path="/" element={<Landing />} />
+      {/* <Routes> */}
+      {/*  <Route path="/" element={<Landing />} /> */}
 
-        <Route path={`${ClientRoutes.Assessment.Landing.path.absolute}/*`} element={<Assessment />} />
+      {/*  <Route path={`${ClientRoutes.Assessment.Landing.path.absolute}/*`} element={<Assessment />} /> */}
 
-        <Route path="*" element={<Navigate to={ClientRoutes.Root.path} replace />} />
-      </Routes>
+      {/*  <Route path="*" element={<Navigate to={ClientRoutes.Root.path} replace />} /> */}
```
Comment (minotogna) [inline src/client/pages/PageRoutes/PageRoutes.tsx]: the should be included now

File: src/client/pages/PageRoutes/PageRoutes.tsx [hunk 4]
```diff
@@ -22,14 +20,15 @@ const PageRoutes: React.FC = () => {
   return (
     <>
       <Toaster />
+      <RouterProvider router={createBrowserRouter(routes)} />
 
-      <Routes>
-        <Route path="/" element={<Landing />} />
+      {/* <Routes> */}
+      {/*  <Route path="/" element={<Landing />} /> */}
```
Comment (minotogna) [inline src/client/pages/PageRoutes/PageRoutes.tsx]: what happened to landing ?

**Comment by minotogna** [comment]
@sorja . post vides not screenshots when dealing with changes please

File: src/client/pages/PageRoutes/PageRoutes.tsx (hunk 4 above)
Comment (sorja) [reply src/client/pages/PageRoutes/PageRoutes.tsx]: Landing was only redirect which was moved to it's own hook

Comment (minotogna) [reply src/client/pages/PageRoutes/PageRoutes.tsx]: better to leave the Page that does that

File: src/client/pages/Cycle/Cycle.tsx [hunk 5]
```diff
@@ -1,71 +1,64 @@
...
-    }
-  }, [userCycles, cycle, navigate, isLogin])
+  // // TODO: think later
+  // useEffect(() => {
+  //   // user has no permission to view cycle
+  //   // If user is accessing login page, do not redirect
+  //   if (!isLogin && !userCycles?.find((userCycle) => userCycle.id === cycle.id)) {
+  //     navigate('/')
+  //   }
+  // }, [userCycles, cycle, navigate, isLogin])
```
Comment (minotogna) [inline src/client/pages/Cycle/Cycle.tsx]: this should be enabled. and the PageLayout must derive itself if it needs the header and toolbar now

File: src/client/pages/Cycle/Cycle.tsx [hunk 6]
```diff
@@ -1,71 +1,64 @@
...
+  //     </Routes>
+  //   </PageLayout>
+  // )
+
+  // return <PageLayout withHeader={!print} withToolbar={!isAdmin && !isLogin && !isUserEditPage} />
 
-  return (
-    <PageLayout withHeader={!print} withToolbar={!isAdmin && !isLogin && !isUserEditPage}>
-      <Routes>
-        <Route path="" element={<AssessmentSwitch components={Components} />} />
```
Comment (minotogna) [inline src/client/pages/Cycle/Cycle.tsx:61]: also ```<Route path="" element={<AssessmentSwitch components={Components} />} />``` must be enabled

File: src/client/pages/CycleLanding/CycleLanding.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+
+import { AssessmentName, AssessmentNames } from 'meta/assessment'
+
+import AssessmentSwitch from 'client/components/AssessmentSwitch'
+import Partners from 'client/components/Partners'
+import Overview from 'client/pages/AssessmentHome/PanEuropeanHome/Overview'
+import Introduction from 'client/pages/Cycle/Introduction'
+import KeyFindings from 'client/pages/Cycle/KeyFindings'
```
Comment (minotogna) [inline src/client/pages/CycleLanding/CycleLanding.tsx]: better to move Partners, Introduction and KeyFindings here.

**Review by minotogna — APPROVED** [review]

**Pull request merged**